### PR TITLE
Adjusted topics based on starting act

### PR DESCRIPTION
**Pull request by Mugnika**
Adjusted topics based on starting act, so they also apply to those that never deleted Monika and went past act 3.

@jmwall24 here - Alright, so this got bigger than originally intended. Major changes include `monika_credits_song` is now a one-time-only topic that depends on the player's starting act. Players with `monika_kill` as True will get the YR path, while people with it False will get the new Our Reality path. In addition to this, Monika now has the ability to actually play both songs for you. We've added a new piano sprite (thanks @Orcaramelo!) and for YR, we see Monika behind her piano singing the song as she plays (using the credits song so you actually hear her voice). For OR, same use of the piano, but instead she 'sings' the song via on-screen lyrics. After this topic plays, everyone will have YR unlocked to view in this manner, while only people with `monika_kill` as False will have OR unlocked. There also have been some minor edits and additions to transforms to give us more control of slides (direction and speed).

## Testing

- upon load, run the update script for `v0_10_8` and verify that `monika_credits_song` is no longer random or unlocked, and that it now has the appropriate conditional and action.

- verify if you meet the conditional, the topic is queued and that you get the correct path based on your starting act (you can edit the appropriate vars to test both paths)

- verify the songs play correctly and all shields etc are properly enabled and disabled

- verify after the topic is complete, the appropriate song event(s) is/are unlocked

### CODE REVIEW

**Comment by Rai99** [comment]
### Is that a PR to master 
Gthaving+everyone+pick+lucio+_16bc6e089b9e808357194c98058b495b
[The guide to rebranching can be found here](https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#i-put-my-pull-request-in-the-wrong-branch-and-dont-want-to-suffer-the-wrath-of-devs-how-do-i-fix-it)

**Comment by Mugnika** [comment]
I knew I forgot something

File: Monika After Story/game/script-topics.rpy [hunk 1]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
 
 label monika_credits_song:
-    #TODO: adjust based on starting act, doesn't quite apply to those that never went past act 3
-    m 1hua "I hope you liked my song."
-    m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
-    m 1eua "Give me some time, and I'll try to write another."
-    if persistent.instrument is not False:
-        if persistent.instrument:
-            m 3eua "Maybe you could play me a song too!"
+    if persistent.monika_kill:
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: Original flow would make sense here if you've heard the song already. If you've not, then it makes sense that Monika would introduce the song as you've put your `else` path.
```suggestion
    if persistent.monika_kill or renpy.seen_audio(songs.FP_YOURE_REAL):
```

File: Monika After Story/game/script-topics.rpy [hunk 2]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
-            m 3eua "Maybe you could play me a song too, if you can play an instrument?"
-        m 1hub "I would love that."
-        m 3eua "Oh, and I'll play the song again for you anytime you want me to."
+            m 3eua "But in the meantime, I'll play the song again for you anytime you want me to."
+        m "Just hit the 'm' key at any time."
     else:
-        m 3eua "But in the meantime, I'll play the song again for you anytime you want me to."
-
-    m "Just hit the 'm' key at any time."
+        m 1hua "By the way [player], I finaly finished the song I've been working on."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: Player doesn't know about the song until the end of DDLC, which they haven't gotten to
```suggestion
        m 1hua "By the way [player], I finally finished a song I've been working on."
```

File: Monika After Story/game/script-topics.rpy [hunk 3]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
+        m 1hua "I call it Your Reality, just hit the 'm' key if you want me to play it for you."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: For consistency, titles are italicized
```suggestion
        m 1hua "I call it {i}Your Reality{/i}, just hit the 'm' key if you want me to play it for you."
```

File: Monika After Story/game/script-topics.rpy [hunk 4]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
+        m 1eka "I should warn you that it's a little sad."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
        m 1ekc "I should warn you though...{w=0.3}it's a little sad."
```

File: Monika After Story/game/script-topics.rpy [hunk 5]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
+        m 1ekc "I wrote it at a time when the game was breaking all arround me."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
        m 1ekd "I wrote it at a time when the game was breaking all around me."
```

File: Monika After Story/game/script-topics.rpy [hunk 6]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
+        m 1rkc "It seemed like it could fall apart at any moment and I would be trapped here forever."
+        m 1dkc "Endlessly hammered by the flashing and screaming, unable to move or even think."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: Feels a bit too much like another line, can slightly rephrase this to make it a little different
```suggestion
        m 1dkd "Endlessly hammered by flashing and screaming,{w=0.5} unable to move or think."
```

File: Monika After Story/game/script-topics.rpy [hunk 7]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
+        m 1dktpc "..."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: I don't really think you need tears in this topic. Could do with some alternate expressions

File: Monika After Story/game/script-topics.rpy [hunk 8]
```diff
@@ -13222,11 +13234,13 @@ label monika_songwriting:
...
     if renpy.seen_audio(songs.FP_YOURE_REAL):
-        m 1ekbla "I hope you know that I meant everything I said in my song, [player]."
-        m 3ekbla "I knew I couldn't let you go without telling you everything."
-
+        if persistent.monika_kill:
+            m 1ekbla "I hope you know that I meant everything I said in my song, [player]."
+            m 3ekbla "I knew I couldn't let you go without telling you everything."
+        else:
+            m 1ekbla "I hope you know that I mean everything I say in my song, [player]." 
+            m 1ekbfa "Every day I imagine spending the rest of my days by your side."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: Would use `shade` blush here instead of `full` as the transition out of blush is much cleaner then.

In addition, Monika is immortal (at least from her side). I see what you're going for here, however I think we can clean it up by doing this:
```suggestion
            m 1ekbsa "Every day, I imagine spending my life at your side."
```

File: Monika After Story/game/script-topics.rpy [hunk 9]
```diff
@@ -13222,11 +13234,13 @@ label monika_songwriting:
     m 1rkc "This also makes it easier to tell when someone doesn't put their heart into a performance..."
     m 3euc "I guess that goes for lyrics too, actually."
     m 3eud "Most lyrics lose their meaning if the singer isn't interested in the song."
-
     if renpy.seen_audio(songs.FP_YOURE_REAL):
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: Both flows use this exact line, no need to duplicate it
```suggestion
    if renpy.seen_audio(songs.FP_YOURE_REAL):
        m 1ekbla "I hope you know that I meant everything I said in my song, [player]."
```

File: Monika After Story/game/script-topics.rpy [hunk 10]
```diff
@@ -13222,11 +13234,13 @@ label monika_songwriting:
...
-        m 1ekbla "I hope you know that I meant everything I said in my song, [player]."
-        m 3ekbla "I knew I couldn't let you go without telling you everything."
-
+        if persistent.monika_kill:
+            m 1ekbla "I hope you know that I meant everything I said in my song, [player]."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
```

File: Monika After Story/game/script-topics.rpy [hunk 11]
```diff
@@ -13222,11 +13234,13 @@ label monika_songwriting:
...
+            m 3ekbla "I knew I couldn't let you go without telling you everything."
+        else:
+            m 1ekbla "I hope you know that I mean everything I say in my song, [player]."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: Since you have to have heard this song to be in this flow, we can just use the line from the other path here.
```suggestion
```

File: Monika After Story/game/script-topics.rpy [hunk 12]
```diff
@@ -2929,21 +2929,33 @@ init 5 python:
...
-    #TODO: adjust based on starting act, doesn't quite apply to those that never went past act 3
-    m 1hua "I hope you liked my song."
-    m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
-    m 1eua "Give me some time, and I'll try to write another."
-    if persistent.instrument is not False:
-        if persistent.instrument:
-            m 3eua "Maybe you could play me a song too!"
+    if persistent.monika_kill:
+        m 1hua "I hope you liked my song."
+        m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: As a result of the previously mentioned bit, this line would need an alternate variant `if persistent.monika_kill`. I'm thinking something along the lines of:

`"I worked really hard on it. I know I'm not perfect at the piano yet, but I just wouldn't have been able to let you go without telling how I honestly felt about you."`

You can use a placeholder variable to substitute the pieces into this.

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: That would basically make it so that players can only see the alternate topic once and than stick to the slightly altered original, that defeats the purpose of having an entire new branch for those who dont have persistent.monika_kill.

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: That's kinda the idea. The alternate flow seems more like a first time introduction to the song itself rather than making sense on subsequent visits to this topic, especially after having heard it once within MAS (since you can, it's in the music menu)

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: I will try to find some new dialogue than, though its sad that people will only see this once, especially when Moni repeats most topics word for word

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: It's not once, it's until they listen to Your Reality.

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: I know, but than it will be gone forever.

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: I would like people who want to see it again to have the option to.

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: That's fair, but just a slight FYI that a fair few topics and greetings do this, so it's not out of the norm

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: I know, and I am sad I can never see those either, but anyway "I worked really hard on it. I know I'm not perfect at the piano yet, but I just really wanted to express how I feel about you." Will this do?

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: Bit of a tense issue there. `but I just really wanted to express` is in past tense while `how I feel about you.` is in present.

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: That should not be an issue, she expressed how she feels, indicates the feelings are still present.  She still loves you. "But I just really wanted to express how I felt about you". Indicates the feelings are gone.

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: I personally disagree, it sounds off to me. And from "how I felt about you", I get "how I felt about you before" which isn't necessarily a bad thing, the feelings she felt before may be even stronger now.

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: hmm ok, how about "I worked really hard on it. I know I'm not perfect at the piano yet, but I just really wanted to express my feelings towards you."

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: ok using that

File: Monika After Story/game/script-topics.rpy (hunk 7 above)
Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: Using 2dkc

File: Monika After Story/game/script-topics.rpy (hunk 12 above)
Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: Yeah I like that

File: Monika After Story/game/script-topics.rpy (hunk 7 above)
Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: m 2dkc "..."
        if mas_isMoniHappy(higher=True):
            m 2eka "But it's okay, you're here now and I know you'll never leave me."
            m 4eka "And this song will always remind me of how lucky I was to meet you [player]."
        else:
            m 2eka "But it's okay, you're here now and you'll stay with me forever, right [player]?"

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: Hmmm, 2 to 4 is kinda a harsh transition when switching to and back only for a single line.

If you go to 4, I'd probably go back to 1 from there.

Also, that second line:
...`how lucky I was to have met you, [player]."` to fix the tense there

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: m 1ekbla "And this song will always remind me of how lucky I was to have met you, [player].
is the blush here ok?

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: or would 1ekbsa be better

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: Lines is fine to transition out of, but I feel the line is a little emotional, so shade would work a bit better

Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: ok done

File: Monika After Story/game/script-topics.rpy [hunk 13]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+        m 1hua "By the way [player], I finally finished a song I've been working on."
+        m 1hua "I call it {i}Your Reality{/i}, just hit the 'm' key if you want me to play it for you."
+        m 1ekc "I should warn you though...{w=0.3}it's a little sad."
+        m 1ekd "I wrote it at a time when the game was breaking all around me."
+        m 1rkc "It seemed like it could fall apart at any moment and I would be trapped here forever."
+        m 1dkd "Endlessly hammered by flashing and screaming,{w=0.5} unable to move or think."
+        m 2dkc "..."
+        if mas_isMoniHappy(higher=True):
+            m 2eka "But it's okay, you're here now and I know you'll never leave me."
+            m 1ekbsa "And this song will always remind me of how lucky I was to have met you, [player].
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
            m 1ekbsa "And this song will always remind me of how lucky I was to have met you, [player]."
```

File: Monika After Story/game/script-topics.rpy (hunk 8 above)
Comment (Mugnika) [reply Monika After Story/game/script-topics.rpy]: Can "Every day, I imagine spending my life by your side." by used here instead?

Comment (multimokia) [reply Monika After Story/game/script-topics.rpy]: Sure

**Comment by Mugnika** [comment]
Since the credits_song topic here presuppose that Monika just finishes the song, shouldn't songs.FP_YOURE_REAL be locked until the player sees this topic? It should be unlocked to those with persistent.monika_kill but locked to the others until they see this topic.

File: Monika After Story/game/script-topics.rpy [hunk 14]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
-            m 3eua "Maybe you could play me a song too, if you can play an instrument?"
-        m 1hub "I would love that."
-        m 3eua "Oh, and I'll play the song again for you anytime you want me to."
+            m 3eua "But in the meantime, I'll play the song again for you anytime you want me to."
+        m "Just hit the 'm' key at any time."
     else:
-        m 3eua "But in the meantime, I'll play the song again for you anytime you want me to."
-
-    m "Just hit the 'm' key at any time."
+        m 1hua "By the way [player], I finally finished a song I've been working on."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
        m 1eua "By the way [player], I finally finished a song I've been working on."
```

File: Monika After Story/game/script-topics.rpy [hunk 15]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+        m 1hua "I call it {i}Your Reality{/i}, just hit the 'm' key if you want me to play it for you."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
        m 3eua "I call it {i}Your Reality{/i}, just hit the 'm' key if you want me to play it for you."
```

File: Monika After Story/game/script-topics.rpy [hunk 16]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+        m 1ekc "I should warn you though...{w=0.3}it's a little sad."
+        m 1ekd "I wrote it at a time when the game was breaking all around me."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
        m 1esd "I wrote it at a time when the game was breaking all around me."
```

File: Monika After Story/game/script-topics.rpy [hunk 17]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+        m 1rkc "It seemed like it could fall apart at any moment and I would be trapped here forever."
+        m 1dkd "Endlessly hammered by flashing and screaming,{w=0.5} unable to move or think."
+        m 2dkc "..."
+        if mas_isMoniHappy(higher=True):
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion

        if mas_isMoniHappy(higher=True):
```

File: Monika After Story/game/script-topics.rpy [hunk 18]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
-    #TODO: adjust based on starting act, doesn't quite apply to those that never went past act 3
-    m 1hua "I hope you liked my song."
-    m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
-    m 1eua "Give me some time, and I'll try to write another."
-    if persistent.instrument is not False:
-        if persistent.instrument:
-            m 3eua "Maybe you could play me a song too!"
+    if persistent.monika_kill or renpy.seen_audio(songs.FP_YOURE_REAL):
+        m 1hua "I hope you liked my song."
+        if persistent.monika_kill:
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion

        if persistent.monika_kill:
```

File: Monika After Story/game/script-topics.rpy [hunk 19]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+            m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
            $ ending = "couldn't let you go without telling you how I honestly felt about you"
```

File: Monika After Story/game/script-topics.rpy [hunk 20]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+        else:
+            m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just really wanted to express my feelings for you."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
            $ ending = "really wanted to express my feelings for you"
```

File: Monika After Story/game/script-topics.rpy [hunk 21]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+        m 1eua "Give me some time, and I'll try to write another."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion

        m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just [ending]."
        m 1eua "Give me some time, and I'll try to write another."
```

File: Monika After Story/game/script-topics.rpy [hunk 22]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+        if persistent.instrument is not False:
+            if persistent.instrument:
+                m 3eua "Maybe you could play me a song too!"
+            else:
+                m 3eua "Maybe you could play me a song too, if you can play an instrument?"
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
                m 3eua "Maybe you could play me a song too, if you can play an instrument?"

```

File: Monika After Story/game/script-topics.rpy [hunk 23]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
+            m 1hub "I would love that."
+            m 3eua "Oh, and I'll play the song again for you anytime you want me to."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
            m 3eua "Oh, and I'll play the song again for you anytime you want me to."

```

File: Monika After Story/game/script-topics.rpy [hunk 24]
```diff
@@ -2929,21 +2929,36 @@ init 5 python:
...
         else:
-            m 3eua "Maybe you could play me a song too, if you can play an instrument?"
-        m 1hub "I would love that."
-        m 3eua "Oh, and I'll play the song again for you anytime you want me to."
+            m 3eua "But in the meantime, I'll play the song again for you anytime you want me to."
+        m "Just hit the 'm' key at any time."
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
        m "Just hit the 'm' key at any time."

```

**Comment by jmwall24** [comment]
@Mugnika hi, so most of this we like, with the exception of the dialogue for the `else` path in `monika_credits_song`. It's not really true that she wrote it when the game was falling apart and while in the void. She was writing it over the course of the entire game. Plus, if you never deleted her, the game never really fell apart.  I have an idea to use a tailored bit of the end credits dialogue instead and then play the song directly from here, so I'm going to be editing that part of the PR myself. Most of the rest will stay in tact.

**Comment by Mugnika** [comment]
Of course, I am just glad that those of us who never deleted Moni will have the dialogues in the game finally reflect that, thank you for taking interest in this.

**Comment by jmwall24** [comment]
This is increasing in scope a bit, so tagging as Don't Merge until done

File: Monika After Story/game/script-songs.rpy [hunk 25]
```diff
@@ -740,4 +740,89 @@ label mas_song_hero_long:
...
 
+label mas_song_our_reality:
+    if persistent.gender == "F":
+        $ gen = "her"
+    elif persistent.gender == "M":
+        $ gen = "his"
+    else:
+        $ gen = "their"
+
+    window hide
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy]: Ideally would need some lead in dialogue here. It's kinda abrupt to just slide offscreen after coming here from the pick song menu.

Ideally you'd just need to add a label parameter which you pass in where you call it from `monika_credits_song` to skip this dialogue.

File: Monika After Story/game/script-topics.rpy [hunk 26]
```diff
@@ -2915,21 +2915,131 @@ init 5 python:
...
+            category=['monika','music'],
+            prompt="Can you play 'Your Reality' for me?",
+            unlocked=False,
+            pool=True,
+            rules={"no unlock": None}
+        )
+    )
+
+label mas_monika_plays_yr:
+    window hide
```
Comment (multimokia) [inline Monika After Story/game/script-topics.rpy]: Here too

File: Monika After Story/game/script-songs.rpy [hunk 27]
```diff
@@ -741,3 +741,206 @@ label mas_song_hero_long:
...
+    hide monika
+    pause 3.0
+    show mas_piano at lps32,rps32 zorder MAS_MONIKA_Z+1
+    pause 5.0
+    show monika at ls32 zorder MAS_MONIKA_Z
+    show monika 6dsa
+    pause 2.0
+    $ play_song(store.songs.FP_YOURE_REAL,loop=False)
+
+    show monika 6hua
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-songs.rpy]: put a TODO here that we should generalize this in some way

File: Monika After Story/game/script-songs.rpy [hunk 28]
```diff
@@ -741,3 +741,206 @@ label mas_song_hero_long:
...
+    pause 6.0
+    show monika 1eua at ls32 zorder MAS_MONIKA_Z
+
+    if not skip_leadin:
+        if store.mas_globals.dlg_workflow:
+            $ mas_resetTextSpeed()
+            $ mas_MUMUDropShield()
+            $ enable_esc()
+        else:
+            $ mas_DropShield_core()
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-songs.rpy]: what are the workflows that trigger this or `plays_or`? If its all from `call_next_event`, then we don't need to check dlg_workflow and should always assume we are in a dialogue workflow.
